extensions/markdown/src/extension.ts

```diff
@@ -0,0 +1,170 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+'use strict';
+
+import * as vscode from 'vscode';
+import * as path from 'path';
+import { ExtensionContext, TextDocumentContentProvider, EventEmitter, Event, Uri, ViewColumn } from "vscode";
+
+const hljs = require('highlight.js');
+const mdnh = require('markdown-it-named-headers');
+const md = require('markdown-it')({
+	html: true,
+	highlight: function (str, lang) {
+		if (lang && hljs.getLanguage(lang)) {
+			try {
+				return `<pre class="hljs"><code><div>${hljs.highlight(lang, str, true).value}</div></code></pre>`;
+			} catch (error) { }
+		}
+		return `<pre class="hljs"><code><div>${md.utils.escapeHtml(str)}</div></code></pre>`;
+	}
+}).use(mdnh, {});
```

Comment:
For markdown-it plugins I suggest that we explore a path via settings. For instance, a user can manually configure additional plugins to use in `settings.json` like `markdown.plugins: ["p1", "p2"]`. We would try to find and configure them, but *not* install them.